src/vivarium/framework/time.py

```diff
@@ -145,11 +145,14 @@ def step_backward(self) -> None:
     def step_forward(self, index: pd.Index) -> None:
         """Advances the clock by the current step size, and updates aligned simulant clocks."""
         self._clock_time += self.step_size
-        if self._individual_clocks:
+        if self._individual_clocks and index.any():
             update_index = self.get_active_simulants(index, self.time)
             clocks_to_update = self._individual_clocks.get(update_index)
             if not clocks_to_update.empty:
                 clocks_to_update["step_size"] = self._step_size_pipeline(update_index)
+                clocks_to_update.loc[
+                    ~clocks_to_update["tracked"], "step_size"
```

Comment:
the point here is to preserve the existing behavior of propagating untracked simulants to events (in case, for example, we want to *re-track* them later on), while excluding them from the logic that determines what the next time to update a simulant is.